tactics/elim.ml

```diff
@@ -32,10 +31,7 @@ let general_elim_using mk_elim (ind, u, args) id =
     let flags = Unification.elim_flags () in
     match mk_elim with
```

Comment:
AFAICT the part of the Goal.enter before this line can be moved to the Elim branch
It may even make sense to have 
~~~ocaml
let general_elim_using mk_elim (ind, u, args) id =
  match mk_elim with
  | Case dep -> Clenv.case_pf ~dep (mkVar id, mkApp (mkIndU (ind, u), args))
  | Elim ->
    Goal.enter ...
~~~
ie move the Elim branch outside this Goal.enter